x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/PivotConfig.java

```diff
@@ -40,14 +43,26 @@
                 args -> {
                     @SuppressWarnings("unchecked")
                     List<GroupConfig> groups = (List<GroupConfig>) args[0];
-                    AggregationConfig aggregationConfig = (AggregationConfig) args[1];
+
+                    // allow "aggs" and "aggregations" but require one to be specified
+                    // for the pathological case somebody specifies both we take "aggregations"
```

Comment:
`DatafeedConfig` takes `aggs` in this case and `AggregatorFactories` throws an error on encountering the second one, so as it stands this code is introducing a third way of doing it.

I think the behaviour of throwing an error if both are specified is best, and it's not very hard to achieve with code very similar to what you have:

```
                        AggregationConfig aggregationConfig = null;
	                    if (args[1] != null) {
	                        aggregationConfig = (AggregationConfig) args[1];
	                    }
                        if (args[2] != null) {
                            if (aggregationConfig == null) {
                                throw new IllegalArgumentException("Found two aggregation definitions: [aggs] and [aggregations]");
                            }
	                        aggregationConfig = (AggregationConfig) args[2];
	                    }
                        if (aggregationConfig == null) {
	                        throw new IllegalArgumentException("Required [aggregations]");
	                    }
```

Reply:
see https://github.com/elastic/elasticsearch/pull/38706/commits/1fe69a3ab0363f3e2a3c33fbb0809efa4e01327b